### PR TITLE
Fix EOF error for downloading zip files

### LINKED ISSUE #4080: MultiModelServing: EOF error when downloading zip files
/kind bug

**What steps did you take and what happened:**
Hello, I am currently testing MultiModelServing and I have encountered the problem that zip files cannot be downloaded as storage URI. I get the following error:
`"Failed to download model model10 with err failed to download model: failed to download model: unable to copy file content: EOF","stacktrace":"github.com/kserve/kserve/pkg/agent.(*Puller).modelProcessor.func1\n\t/go/src/github.com/kserve/kserve/pkg/agent/puller.go:156\ngithub.com/kserve/kserve/pkg/agent.(*Puller).modelProcessor\n\t/go/src/github.com/kserve/kserve/pkg/agent/puller.go:220"}`

**What did you expect to happen:**
zip file completely copied and unzipped under /mnt/models

**What's the InferenceService yaml:**
```
[apiVersion: "serving.kserve.io/v1beta1"
kind: "InferenceService"
metadata:
  name: "custom2"
spec:
  predictor:
    model:
      modelFormat:
        name: sklearn
      protocolVersion: v2
      runtime: custom-runtime
      resources:
        requests:
          cpu: "1"         
          memory: "16Gi"
          ephemeral-storage: "16Gi"
        limits:
          cpu: "1"
          memory: "16Gi"
          ephemeral-storage: "16Gi"
---
apiVersion: "serving.kserve.io/v1alpha1"
kind: "TrainedModel"
metadata:
  name: "model1"
spec:
  inferenceService: "custom2"
  model:
    storageUri: "https://www.abc.net/v1.zip"
    framework: "test"
    memory: "1Gi"
```

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Istio Version:
- Knative Version:
- KServe Version: 0.14.0
- Kubeflow version: 1.9.1
- Cloud Environment:[k8s_istio/istio_dex/gcp_basic_auth/gcp_iap/aws/aws_cognito/ibm]
- Minikube/Kind version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by Jonas-Bruns**
**What this PR does / why we need it**:
This PR fixed the problem that zip files are not downloaded correctly to /mnt/models.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #4080

**Type of changes**
Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note

```

**Re-running failed tests**

- `/rerun-all` - rerun all failed workflows.
- `/rerun-workflow <workflow name>` - rerun a specific failed workflow. Only one workflow name can be specified. Multiple /rerun-workflow commands are allowed per comment.

### CODE REVIEW

File: pkg/agent/storage/https.go [hunk 1]
```diff
@@ -246,7 +246,7 @@ func extractTarFiles(reader io.Reader, dest string) error {
 		}
 
 		// gosec G110
-		_, err := io.CopyN(newFile, tr, DEFAULT_MAX_DECOMPRESSION_SIZE)
+		_, err = io.CopyN(newFile, tr, DEFAULT_MAX_DECOMPRESSION_SIZE)
```
Comment (spolti) [inline pkg/agent/storage/https.go]: maybe it is better to assign the error to a new env.
```suggestion
		_, ioErr := io.CopyN(newFile, tr, DEFAULT_MAX_DECOMPRESSION_SIZE)
```

Comment (Jonas-Bruns) [reply pkg/agent/storage/https.go]: Changed.

**Comment by sivanantha321** [comment]
@Jonas-Bruns Can you help fix the lint error ?

**Comment by Jonas-Bruns** [comment]
> @Jonas-Bruns Can you help fix the lint error ?

Should be fixed. Unfortunately I can't run the linter locally because it produces other errors

**Review by terrytangyuan — APPROVED** [review]
Thanks!

/lgtm
/approve

**Pull request merged**